crates/phactory/src/prpc_service.rs

```diff
@@ -927,6 +969,155 @@ impl<Platform: pal::Platform + Serialize + DeserializeOwned> Phactory<Platform>
             .load_proof(proof);
         Ok(())
     }
+
+    fn save_cluster_state(
+        &self,
+        min_block_number: BlockNumber,
+        receiver: &[u8],
+    ) -> RpcResult<pb::SaveClusterStateResponse> {
+        let Ok(receiver) = receiver.try_into() else {
+            return Err(from_display("Invalid receiver"));
+        };
+        let receiver = WorkerPublicKey(receiver);
+        let Some(system) = &self.system else {
+            return Err(from_display("System is uninitialized"));
+        };
+        let Some(runtime_state) = &self.runtime_state else {
+            return Err(from_display("Runtime is uninitialized"));
+        };
+        let Some(cluster) = &system.contract_cluster else {
+            return Err(from_display("No cluster found"));
+        };
+        let block_number = system.block_number;
+        if block_number < min_block_number {
+            return Err(from_display(format!(
+                "Can not save cluster state at block {block_number} which earlier than \
+                 min={min_block_number}",
+            )));
+        }
+        if min_block_number + 128 < block_number {
```

Comment:
128 blocks are around 25 mins, which is very reasonable. How about documenting it somewhere?

Reply:
Added a note in `add-worker-to-cluster.md`.